src/transformers/models/dallemega/modeling_dallemega.py

```diff
@@ -0,0 +1,1557 @@
+            attention_mask = torch.cat([attention_mask, attention_mask_uncond], dim=0)
+
+        _, seq_len = input_shape[:2]
+        positions_ids = torch.arange(0, seq_len, dtype=torch.long, device=inputs_embeds.device)
+        embed_pos = self.embed_positions(positions_ids)
+
+        hidden_states = inputs_embeds + embed_pos
+        hidden_states = self.layernorm_embedding(hidden_states)
+
+        # expand attention_mask
+        if attention_mask is not None:
+            # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
+            attention_mask = _expand_mask(attention_mask, inputs_embeds.dtype)
+
+        encoder_states = () if output_hidden_states else None
+        all_attentions = () if output_attentions else None
+
+        # check if head_mask has a correct number of layers specified if desired
+        if head_mask is not None:
+            if head_mask.size()[0] != (len(self.layers)):
+                raise ValueError(
+                    f"The head_mask should be specified for {len(self.layers)} layers, but it is for"
+                    f" {head_mask.size()[0]}."
+                )
+
+        for idx, encoder_layer in enumerate(self.layers):
+            if output_hidden_states:
+                encoder_states = encoder_states + (hidden_states,)
+
+            if self.gradient_checkpointing and self.training:
+
+                def create_custom_forward(module):
+                    def custom_forward(*inputs):
+                        return module(*inputs, output_attentions)
+
+                    return custom_forward
+
+                layer_outputs = torch.utils.checkpoint.checkpoint(
+                    create_custom_forward(encoder_layer),
+                    hidden_states,
+                    attention_mask,
+                    (head_mask[idx] if head_mask is not None else None),
+                )
+            else:
+                layer_outputs = encoder_layer(
+                    hidden_states,
+                    attention_mask,
+                    layer_head_mask=(head_mask[idx] if head_mask is not None else None),
+                    output_attentions=output_attentions,
+                )
+
+            hidden_states = layer_outputs[0]
+
+            if output_attentions:
+                all_attentions = all_attentions + (layer_outputs[1],)
+
+        hidden_states = self.final_layernorm(hidden_states)
+
+        if output_hidden_states:
+            encoder_states = encoder_states + (hidden_states,)
+        
+        # filter out the unconditional hidden states from encoder_states
+        if do_superconditioning and output_hidden_states:
+            encoder_states = (state.chunk(2)[0] for state in encoder_states)
+        
+        # filter out the unconditional attentions from all_attentions
+        if do_superconditioning and output_attentions:
+            all_attentions = (attn.chunk(2)[0] for attn in all_attentions)
```

Comment:
filter out the unconditional `hidden_states` and `attentions` since we don't want to return those.